addon-api/content-script/Tab.js

```diff
@@ -53,11 +53,40 @@ export default class Tab extends Listenable {
    */
   loadScript(url) {
     return new Promise((resolve, reject) => {
-      const script = document.createElement("script");
-      script.src = url;
-      document.head.appendChild(script);
-      script.onload = resolve;
-      script.onerror = reject;
+      if (scratchAddons.loadedScripts[url]) {
+        const obj = scratchAddons.loadedScripts[url];
+        if (obj.loaded) {
+          // Script has been already loaded
+          resolve();
+        } else if (obj.error === null) {
+          // Script has been appended to document.head, but not loaded yet.
+          obj.script.addEventListener("load", (e) => {
```

Comment:
Won't this wait indefinitely if this errors?

Reply:
Fixed